pkg/resmgr/lib/memory/zones.go

```diff
@@ -154,6 +159,27 @@ func (a *Allocator) zoneFree(zone NodeMask) int64 {
 	return a.zoneCapacity(zone) - a.zoneUsage(zone)
 }
 
+func (a *Allocator) zoneAvailable(zone NodeMask) int64 {
+	available := a.zoneFree(zone)
+	if available <= 0 {
+		return 0
+	}
+
+	// Cap available amount to the smallest available free in any of our ancestors.
+	for _, z := range a.zones {
+		if z.nodes != zone && (z.nodes&zone) == zone {
+			if free := a.zoneFree(z.nodes); free < available {
+				available = free
+			}
+		}
+		if available == 0 {
```

Comment:
Should we worry about available < 0 case here, too, like in line 164?

Reply:
Yes, does not make sense to return anything < 0 here...